src/Vegetation/Vegetation.jl

```diff
@@ -0,0 +1,511 @@
+will be used for evolving the system in time,
+and that the array-like objected being stepped forward
+is a `ClimaCore.Fields.FieldVector`. 
+While a simple array may be sufficient for vegetation models,
+the `FieldVector` type is used in order to make use of 
+`ClimaCore` functionality when solving PDEs (required for
+other components of Land Surface Models) and for ease of
+handling multi-column models.
+To simulate land surfaces with multiple components (vegetation,
+soil, rivers, etc), the ClimaLSM.jl package should be used.
+That package will use the methods of this function for advancing
+the system forward in time, extending methods as needed to account
+for interactions between components.
+=#
+using ClimaLSM
+using ClimaCore
+using UnPack
+using DocStringExtensions
+import ClimaCore: Fields
+
+using ClimaLSM.Domains: AbstractVegetationDomain, VegetationDomain
+import ClimaLSM:
+    AbstractModel,
+    initialize_prognostic,
+    make_update_aux,
+    make_rhs,
+    make_ode_function,
+    prognostic_vars,
+    prognostic_types,
+    auxiliary_vars,
+    initialize,
+    initialize_auxiliary,
+    name
+export VegetationModel,
+    AbstractVegetationModel,
+    flux,
+    effective_saturation_vegetation,
+    augmented_liquid_fraction,
+    van_Genuchten_volume_to_pressure,
+    van_Genuchten_pressure_to_volume,
+    volume_to_pressure,
+    pressure_to_volume,
+    flux_out_roots,
+    VegetationParameters,
+    PrescribedSoilPressure,
+    PrescribedRootFlux,
+    PrescribedTranspiration,
+    AbstractRootExtraction
+
+"""
+    AbstractVegetationModel{FT} <: AbstractModel{FT}
+An abstract type for vegetation models.
+Concrete types include a plant hydraulics model, but future types will
+include multi-layer canopy models and possibly a big leaf model.
+"""
+abstract type AbstractVegetationModel{FT} <: AbstractModel{FT} end
+
+ClimaLSM.name(::AbstractVegetationModel) = :vegetation
+ClimaLSM.domain(::AbstractVegetationModel) = :surface
+
+"""
+    AbstractRootExtraction{FT <: AbstractFloat}
+An abstract type for types representing different models of
+water exchange between soil and plants.
+Currently, only a prescribed soil pressure is supported for
+standalone plant hydraulics. Use within an LSM requires
+types defined within ClimaLSM, and include a prognostic
+soil pressure for models with both soil and roots.
+"""
+abstract type AbstractRootExtraction{FT <: AbstractFloat} end
+
+"""
+    AbstractTranspiration{FT <: AbstractFloat}
+An abstract type for types representing different models of
+transpiration.
+Currently, only a PrescribedTranspiration is supported.
+"""
+abstract type AbstractTranspiration{FT <: AbstractFloat} end
+
+
+"""
+    VegetationParameters{FT <: AbstractFloat}
+A struct for holding parameters of the Vegetation Model.
+$(DocStringExtensions.FIELDS)
+"""
+struct VegetationParameters{FT <: AbstractFloat, PSE}
+    "controls the shape and steepness of conductivity vs. absolute pressure curve, for roots: unitless"
```

Comment:
Why do we need to supply both vG params and the a/b parameters? Wouldn't it make more sense just to use one model?

Reply:
I remember you had brought this up before and it was a good point I thought too, so I did some reading and think there isn't a physical link between K(P) and the vartheta to P function as far as I understood from van G's 1980 paper; so we don't have to use the same model for these two equations. But it's better to have fewer parameters like you say... and so to use one if we can

My only hesitation is I don't know if the vg K(P) function is integrable (I haven't checked but will when getting back to the writing). If it isn't then we should stick with the modified logistic as we rly need an integrable K(P) function (I don't know of a consist p to theta one to go with it, maybe we can find one with same technique vg used but other way around as they found k(p) from a given p to theta I think but we need p to theta given a k(p)) - anyhow, to start off with modified logistic for K(P), I was looking for some params that I knew were plausible so I fit the modified logistic curve Yujie found to Vg with the default ones we had in the soil model

Reply:
Isee - but I stilll dont think we should be using two sets of parameters - especially if finding one from the other requires fitting a curve.

if K(P) needs to be integrable, what if we use the Brooks and Corey formulation then, instead of vG, for both K(theta) and P(theta)?

Alternatively, we can use an approximation of \int K(p) dp. E.g. 1/2 (P2-P1)*K(P2).  or (K(P1) + K(P2))/2*(P2-P1). We are discretizing the starting equation, and incurring error akin to this from the start, so it's not clear to me that we need this integral to be integrable...

Reply:
Hey, sounds good. I will implement (K(P1) + K(P2))/2(P2-P1) with vg as I think easier than integrating Brooks and Corey, and vG params are more common than Brooks and Corey I think, plus more consistent with soil model